emily/lambda/src/errors.rs

```diff
@@ -0,0 +1,166 @@
+            EmilyApiError::Service(Err(ref err)) => Some(err.as_ref()),
+            _ => None,
+        }
+    }
+}
+
+impl EmilyApiError {
+    fn status_code(&self) -> u16 {
+        match *self {
+            EmilyApiError::BadRequest(_) => http::StatusCode::BAD_REQUEST,
+            // EmilyApiError::Forbidden(_) => http::StatusCode::FORBIDDEN,
+            // EmilyApiError::NotFound(_) => http::StatusCode::NOT_FOUND,
+            // EmilyApiError::Conflict(_) => http::StatusCode::CONFLICT,
+            // EmilyApiError::NotImplemented(_) => http::StatusCode::NOT_IMPLEMENTED,
+            // EmilyApiError::Throttling(_) => http::StatusCode::TOO_MANY_REQUESTS,
+            EmilyApiError::Service(_) => http::StatusCode::INTERNAL_SERVER_ERROR,
+        }.as_u16()
+    }
+
+    fn response_body(&self) -> Result<String, serde_json::Error> {
+        match *self {
+            EmilyApiError::BadRequest(_) =>
+                serde_json::to_string(&models::BadRequestErrorResponseContent { message: self.to_string() }),
+            // EmilyApiError::Forbidden(_) =>
+            //     serde_json::to_string(&ForbiddenErrorResponseContent { message: self.to_string() }),
+            // EmilyApiError::NotFound(_) =>
+            //     serde_json::to_string(&NotFoundErrorResponseContent { message: self.to_string() }),
+            // EmilyApiError::Conflict(_) =>
+            //     serde_json::to_string(&ConflictErrorResponseContent { message: self.to_string() }),
+            // EmilyApiError::NotImplemented(_) =>
+            //     serde_json::to_string(&NotImplementedErrorResponseContent { message: self.to_string() }),
+            // EmilyApiError::Throttling(_) =>
+            //     serde_json::to_string(&ThrottlingErrorResponseContent { message: self.to_string() }),
+            EmilyApiError::Service(_) =>
+                serde_json::to_string(&models::ServiceErrorResponseContent { message: self.to_string() }),
+        }
+    }
+
+    #[allow(clippy::wrong_self_convention)]
+    pub fn to_apigw_response(self) -> ApiGatewayProxyResponse {
+        match self.response_body() {
+            Ok(body) => common::SimpleApiResponse {
+                status_code: self.status_code(),
+                body: Some(Body::Text(body)),
+            },
+            // This occurs in the rare case that the API Error itself failed to serialize.
+            Err(err) => common::SimpleApiResponse {
+                status_code: http::StatusCode::INTERNAL_SERVER_ERROR.as_u16(),
+                // This output won't be of the right format for the client, but this is an edge case where
+                // the serialization will have needed to fail on a well defined structure, and this error
+                // can be handled by someone receiving data from the client since they'll need to handle
+                // potential errors anyway.
+                body: Some(Body::Text(format!("Error Deserializing: {}, {}", self, err).to_string())),
```

Comment:
I would generate the body manually with this:
```rust
body: Some(Body::Text(serde_json::json!({"message": format!("SERVER ERROR {err}")}).to_string()))
```
Well, I'd break the above into multiple lines but it would equivalent the above.

Reply:
Of course, this shouldn't ever be an issue, but still.

Reply:
The usage of `serde_json` is better, I'll change that.